app/models/concerns/acts_as_linked_list.rb

```diff
@@ -36,28 +36,51 @@ def list_item_name(name)
     #   the new next items.
     #
     # item - The item to be moved.
+    # parent - The item's parent
     # new_position - The hash that contains the new previous and/or new next
     #   item using the keys :prev_item and :next_item, respectively.
     #
-    # Returns true.
-    def move(item, new_position = {})
+    # Returns true when item is moved successfully to the new position.
+    # Returns false when the new position is invalid.
+    def move(item, parent, new_position = {})
       item.class.transaction do
         previous_item = item.send("previous_#{item.class.name.downcase}")
         next_item = item.send("next_#{item.class.name.downcase}")
+        new_previous = new_position[:prev_item]
+        new_next = new_position[:next_item]
+
+        # Validates new position
+        is_new_position_valid = if new_previous.present?
+                                  next_item_of_new_previous = new_previous.send("next_#{new_previous.class.name.downcase}")
+                                  if next_item_of_new_previous
+                                    new_next == next_item_of_new_previous
+                                  else
+                                    new_next.nil?
+                                  end
+                                else
+                                  if parent.items.empty?
```

Comment:
This still hasn't addressed [this comment](https://github.com/dradis/dradis-ce/pull/1088#discussion_r1053775107). If `parent.items.empty?` resolves to `true`, then `if parent.first_item` will also always resolve to false, no? I added a suggestion for a change in that comment linked above

Reply:
Aaah makes sense. I will do that change